src/lib/FeeEscalatorLib.sol

```diff
@@ -0,0 +1,76 @@
+import {FixedPointMathLib} from "solmate/src/utils/FixedPointMathLib.sol";
+import {Input} from "../base/ReactorStructs.sol";
+import {FeeEscalator, Input} from "../base/ReactorStructs.sol";
+
+library FeeEscalatorLib {
+    using FixedPointMathLib for uint256;
+
+    string public constant FEE_ESCALATOR_TYPESTRING =
+        "FeeEscalator(address token,uint256 startAmount,uint256 endAmount,uint256 startTime,uint256 endTime)";
+    bytes32 internal constant FEE_ESCALATOR_TYPEHASH =
+        keccak256("FeeEscalator(address token,uint256 startAmount,uint256 endAmount,uint256 startTime,uint256 endTime)");
+
+    /// @notice thrown if the escalation direction is incorrect
+    error InvalidAmounts();
+    /// @notice thrown if the end time is before the start time
+    error EndTimeBeforeStartTime();
+
+    /// @notice calculates an amount on a linear curve over time from startTime to endTime
+    /// @dev handles only increasing amounts from startAmount to endAmount
+    /// @param startAmount The amount of tokens at startTime
+    /// @param endAmount The amount of tokens at endTime
+    /// @param startTime The time to start escalating linearly
+    /// @param endTime The time to stop escalating linearly
+    function resolve(uint256 startAmount, uint256 endAmount, uint256 startTime, uint256 endTime)
+        internal
+        view
+        returns (uint256 resolvedAmount)
+    {
+        if (startAmount > endAmount) {
+            revert InvalidAmounts();
+        } else if (endTime < startTime) {
+            revert EndTimeBeforeStartTime();
+        } else if (endTime <= block.timestamp) {
+            resolvedAmount = endAmount;
+        } else if (startTime >= block.timestamp) {
+            resolvedAmount = startAmount;
+        } else {
+            unchecked {
+                uint256 elapsed = block.timestamp - startTime;
+                uint256 duration = endTime - startTime;
+                resolvedAmount = startAmount + (endAmount - startAmount).mulDivDown(elapsed, duration);
+            }
+        }
+    }
+
+    /// @notice Transforms the fee data into a TokenPermissions struct needed for the permit call.
+    function toTokenPermissions(FeeEscalator memory fee)
+        internal
+        pure
+        returns (ISignatureTransfer.TokenPermissions memory permission)
+    {
+        permission = ISignatureTransfer.TokenPermissions({token: fee.token, amount: fee.endAmount});
+    }
+
+    /// @notice Transforms the fee data into the the resolved amount and the provided recipient.
+    function toTransferDetails(FeeEscalator memory fee, address feeRecipient)
+        internal
+        view
+        returns (ISignatureTransfer.SignatureTransferDetails memory detail)
+    {
+        uint256 resolvedAmount = resolve(fee.startAmount, fee.endAmount, fee.startTime, fee.endTime);
+        detail = ISignatureTransfer.SignatureTransferDetails({to: feeRecipient, requestedAmount: resolvedAmount});
```

Comment:
I think the user should still have to sign address(0) to signify that any filler can fill/that there can be any fee recipient. otherwise the fee recipient should be the one the user signed.

i think this means we need to add a recipient field to the FeeEscalator struct

Reply:
Agree, we should keep the behavior on main where an order _could_ be made for a specific filler